examples/with-next/app/mockProvider.tsx

```diff
@@ -0,0 +1,31 @@
+'use client'
+import { useEffect, useState } from 'react'
+
+export function MockProvider({
```

Comment:
How about using suspense?

mockProvider.tsx

```tsx
'use client'

let triggered = false

async function enableApiMocking() {
  const { worker } = await import('../mocks/browser')
  await worker.start()
}

export function MockProvider() {
  if (!triggered) {
    triggered = true
    throw enableApiMocking()
  }

  return null
}
```

layout.tsx

```tsx
export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode
}>) {
  return (
    <html lang="en">
      <body className={inter.className}>
        <MockProvider />
        {children}
      </body>
    </html>
  )
}
```

By doing so, we can avoid wrapping children in the mock provider client component.
But I am not sure if this is a good solution.

useEffect|Suspense
-|-
<img width="458" alt="ss2" src="https://github.com/mswjs/examples/assets/1910945/ede7c1e4-d5ea-45ae-8685-a2fe6da86234">|<img width="546" alt="ss" src="https://github.com/mswjs/examples/assets/1910945/b2773c2a-220a-4e80-8f0e-898e06309f80">

Reply:
The goal of this is to defer the rendering of the children until the service worker is activated. You are proposing keeping the state internally but I don't see it affecting `{children}`. So they will render, and if they make any HTTP requests, those will not be intercepted because the worker is not ready yet.